src/Symfony/Component/Workflow/Workflow.php

```diff
@@ -59,10 +59,6 @@ public function getMarking($subject)
     {
         $marking = $this->markingStore->getMarking($subject);
 
-        if (!$marking instanceof Marking) {
```

Comment:
I guess we're not gonna agree on this. However i understand the concerns when it comes to "missing the contract and debugging", but the exception would be cosmetic only. Ie. the plain error you get 9/10 is just as informative (stacktrace wise), eventually you'll have to debug your implementation anyway. And if error handling is poor, i would advice fixing that.

> is it common for users to write their own implementation of the interface

From a symfony/framework pov this is almost any interface... which is exactly why i think we have to draw the line on this.

For debugging/validating on return types; i think this should be be mostly covered by static code analysis, tests, and eventually PHP7.

This is way i changed it. :)